address_translators.go

```diff
@@ -0,0 +1,53 @@
+package gocql
+
+import (
+	"log"
+	"net"
+)
+
+// AddressTranslator provides a way translate node addresses (and ports) when one is
+// discovered or received as a node event. This is especially useful in ec2 (when
+// using the EC2MultiRegionAddressTranslator) to translate public IPs to private IPs
+// when possible.
+type AddressTranslator interface {
+	// Translate will translate the provided address and/or port when a host is discovered.
+	Translate(addr string, port int) (string, int)
```

Comment:
I'm trying to ensure that any issue with translation ends up returning what was provided. This also seems in keeping with the other translator implementations.